src/components/templates/PartyMap/components/RoomModalOngoingEvent/RoomModalOngoingEvent.tsx

```diff
@@ -65,7 +67,7 @@ export const RoomModalOngoingEvent: React.FC<RoomModalOngoingEventProps> = ({
         className="btn btn-primary room-entry-button"
         onClick={onRoomEnter}
       >
-        {joinButtonText ?? "Enter"}
+        <a href={roomUrl}>{joinButtonText ?? "Enter"}</a>
       </button>
```

Comment:
I don't believe nesting an `<a href` tag inside a `<button>` like this is valid HTML.

- https://www.reddit.com/r/webdev/comments/ce4w59/is_it_bad_practice_to_wrap_a_button_in_an_a_href/
  - > It's a violation of the HTML5 specification to include either an anchor tag inside a button tag, or a button tag inside an anchor tag. Interactive elements are not permitted to be children of one another, as it introduces ambiguity with regard to user intent.

Reply:
From a quick skim of the code, it also seems like if we were to replace the `<button>` with an `<a href` outright, we may run into some potential issues with the `enterRoomWithSound` feature:

- https://github.com/sparkletown/sparkle/search?q=onRoomEnter

And would need to change the logic around how `enterRoom` works currently, to provide the appropriate props/etc to open internal vs external links:

- https://github.com/sparkletown/sparkle/blob/staging/src/hooks/useRoom.ts#L39-L45

Reply:
thanks for the HTML5 guidance + description of interaction with onRoomEnter.  Personally I do not care about the "Sound" part but it seems that some additional logic is in place indeed, but it is primarily (userId could and probably checked anyways when entering the room) is to switch between `enterVenue(roomVenue.id)` and  `enterExternalRoom`.  

<details>
<summary>relevant code snippets</summary>

```javascript
export const openUrl = (url: string) => {
  if (!isValidUrl(url)) { 
    Bugsnag.notify(
      // new Error(`Invalid URL ${url} on page ${window.location.href}; ignoring`),
      new Error(
        `Invalid URL ${url} on page ${window.location.href}; allowing for now (workaround)`
      ),
      (event) => {
        event.addMetadata("context", { func: "utils/url::openUrl", url });
      }
    );
    // @debt keep the checking in place so we can debug further, but don't block attempts to open
    // return;
  }

  if (isExternalUrl(url)) {
    window.open(url, "_blank", "noopener,noreferrer");
  } else {
    // @debt Possibly use react router here with window.location.pathname.
    window.location.href = url;
  }
};
```

```javascript
export const enterVenue = (venueId: string) => openUrl(venueInsideUrl(venueId));
```

```javascript
export const enterExternalRoom = ({
  userId,
  locationName,
  roomUrl,
}: EnterExternalRoomProps) => {
  setLocationData({
    userId,
    locationName,
  });

  openRoomUrl(roomUrl);
};
```
</details>

so in both cases -- "go to url  (with possible extra guards for external + check and reporting on bad ones) + do some accounting" seems to me.  As for accounting, see my other comment:  seems needs to be fixed up anyways and become more "event driven" than "trying to track the steps". As for needed `<a` attributes for external links -- could be done there in place if information on either it is external or not passed in.  As for checks -- could be done "upstairs" or also just right there, ideally absorbing 


<details>
<summary>following code into a reusable `checkUrl(url)` or alike:</summary> 

```shell
  if (!isValidUrl(url)) { 
    Bugsnag.notify(
      // new Error(`Invalid URL ${url} on page ${window.location.href}; ignoring`),
      new Error(
        `Invalid URL ${url} on page ${window.location.href}; allowing for now (workaround)`
      ),
      (event) => {
        event.addMetadata("context", { func: "utils/url::openUrl", url });
      }
    );
    // @debt keep the checking in place so we can debug further, but don't block attempts to open
    // return;
  }
```
</details>

Reply:
re enterRoomWithSound ... is "enter with sound" a property of the target room  to enter or a property of a location from where to enter?  
if property of the room to enter -- sound could be just played whenever you enter room regardless how you got there (e.g. following a URL).  If it is property of location from which to enter: I wonder if some "onclick" could event be assigned to `a` or may be any room url could handle a `?entersound` option, so where desired -- passed via url.

Reply:
> re enterRoomWithSound ... is "enter with sound" a property of the target room to enter or a property of a location from where to enter?

In our current taxonomy/terminology (simplified), there is a `map`, which has `rooms` (aka: `portals`), that link to `venues`.

When you click on a `room` on the `map`, the `RoomModal` is shown, which has a button to 'enter the room' (which is really 'entering the venue', technically speaking)

`enterRoomWithSound` is intended to play when that 'enter room/venue' button is clicked.

---

> I wonder if some "onclick" could event be assigned to a

I believe this is possible, though i'm not aware off the top of my head if it's 'bad practice'. Semantically, (usually), `<a href` is used for links and such, and `<button>`s are used for adding various JavaScript event handlers when clicked.

- https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a
  - https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#properties
    - It seems like there is an ARIA role for `button` that can be used here.. so that might be what we want
      - https://w3c.github.io/aria/#button
  - https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#onclick_events
    - This is under the 'Accessibility' heading, regarding `onClick` events